hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -647,26 +650,38 @@ public void createDirectory(final Path path, final FsPermission permission, fina
   public AbfsInputStream openFileForRead(final Path path,
       final FileSystem.Statistics statistics)
       throws AzureBlobFileSystemException {
-    return openFileForRead(path, Optional.empty(), statistics);
+    return openFileForRead(path, new OpenFileParameters(), statistics);
   }
 
   public AbfsInputStream openFileForRead(final Path path,
-      final Optional<Configuration> options,
+      final OpenFileParameters parameters,
       final FileSystem.Statistics statistics)
       throws AzureBlobFileSystemException {
     try (AbfsPerfInfo perfInfo = startTracking("openFileForRead", "getPathStatus")) {
       LOG.debug("openFileForRead filesystem: {} path: {}",
               client.getFileSystem(),
               path);
 
+      String resourceType;
+      long contentLength;
+      String eTag;
       String relativePath = getRelativePath(path);
+      Configuration options = null;
 
-      final AbfsRestOperation op = client.getPathStatus(relativePath, false);
-      perfInfo.registerResult(op.getResult());
+      try {
+        options = parameters.getOptions();
+        FileStatus fileStatus = parameters.getStatus();
```

Comment:
fileStatus can be null. If null should do line 678, 679. Then u will have a valid filestatus if the getPathStatus call did not fail. 
U can then set resourcetype, contentlenght and etgag from the filestatus.

Reply:
done